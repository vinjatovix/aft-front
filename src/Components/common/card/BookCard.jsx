import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import { ListItem } from "../ListItem";

export const BookCard = ({ data, isAdmin, actions, token }) => {
  const defaultCoverBook =
    "https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg";
  return (
    <div className="flex-wrapper column">
      <ul
        className="book"
        onClick={() => {
          actions.detail();
        }}
      >
        {<img src={data.img ? data.img : defaultCoverBook} alt={data.name} />}
        <ListItem name="Obra" children={data.name} />
        <ListItem name="Autor" className="small" children={data.author} />
      </ul>
      <NavLink to={`/characters/book/${data._id}`}>
        <button data-testid="char-button" className="bg-blue">
          🎭 Personajes
        </button>
      </NavLink>
      {isAdmin && (
        <div className="control-panel-mini">
          <button
            className="bg-yellow"
            data-testid="edit-button"
            onClick={() => {
              actions.edit();
            }}
          >
            📝 editar
          </button>
          <button
            data-testid="delete-button"
            className="bg-red"
            onClick={() => {
              actions.delete(token, data._id);
            }}
          >
            🗑️ eliminar
          </button>
        </div>
      )}
    </div>
  );
};

BookCard.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    img: PropTypes.string,
  }).isRequired,
  isAdmin: PropTypes.bool,
  actions: PropTypes.object.isRequired,
  token: PropTypes.string.isRequired,
};

BookCard.defaultProps = {
  isAdmin: false,
};
