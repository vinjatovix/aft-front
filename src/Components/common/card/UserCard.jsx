import React from "react";
import PropTypes from "prop-types";

import { ListItem } from "../ListItem";
import { UserRole } from "../../ui/UserRole";

export const UserCard = ({ data }) => {
  return (
    <ul className="user-list">
      <ListItem name="Nombre" children={data.username} />
      <ListItem name="Role" children={UserRole(data.roles)} />
    </ul>
  );
};

UserCard.propTypes = {
  data: PropTypes.shape({
    username: PropTypes.string.isRequired,
    roles: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};
