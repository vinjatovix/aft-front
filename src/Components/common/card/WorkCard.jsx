import React from "react";
import PropTypes from "prop-types";
import { ListItem } from "../ListItem";

export const WorkCard = ({ data }) => {
  return (
    <ul className="work-list">
      <ListItem
        name="Intérprete"
        children={<p>by: {data.metadata.createdBy}</p>}
        liKey={data.metadata.createdBy}
      />
      <ListItem
        name="Obra"
        children={<p>Obra: {data.character.book.name}</p>}
        liKey={data.character.book._id}
      />
      <ListItem
        name="Personaje"
        children={<p>Personaje: {data.character.name}</p>}
        liKey={data.character._id}
      />
      <ListItem
        name="Escena"
        children={<p>Escena: {data.scene.name}</p>}
        liKey={data.scene._id}
      />
      <ListItem
        name="Descripción"
        children={<p>Descripción: {data.description}</p>}
      />
    </ul>
  );
};

WorkCard.propTypes = {
  data: PropTypes.object.isRequired,
};
