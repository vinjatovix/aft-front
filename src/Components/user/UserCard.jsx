import React from "react";
import PropTypes from "prop-types";

import { ListItem } from "../common/ListItem";
import { UserRole } from "../ui/UserRole";

export const UserCard = ({ data: { username, roles } }) => {
  return (
    <ul className="user-list">
      <ListItem name="username" children={username} />
      <ListItem name="role" children={UserRole({ roles })} />
    </ul>
  );
};

UserCard.propTypes = {
  data: PropTypes.shape({
    roles: PropTypes.arrayOf(PropTypes.string).isRequired,
    username: PropTypes.string.isRequired,
  }).isRequired,
};
