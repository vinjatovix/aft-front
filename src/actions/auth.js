import { ACTION_TYPES } from "../types/types";

export const loginRequestAction = () => ({
  type: ACTION_TYPES.AUTH_LOGIN_REQUEST,
});

export const loginSuccessAction = (auth) => ({
  type: ACTION_TYPES.AUTH_LOGIN_SUCCESS,
  payload: auth,
});

export const loginFailureAction = (error) => ({
  type: ACTION_TYPES.AUTH_LOGIN_FAILURE,
  payload: error,
});

export const logoutAction = () => ({
  type: ACTION_TYPES.AUTH_LOGOUT,
});
