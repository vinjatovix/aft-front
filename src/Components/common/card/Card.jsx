import React from "react";
import PropTypes from "prop-types";
import { UserCard } from "./UserCard";

const Card = ({ data, type, actions, isAdmin, token }) => {
  return (
    <div className="card" onClick={() => actions.setDataDetail(data)}>
      {type === "user" && <UserCard data={data} />}
    </div>
  );
};

Card.propTypes = {
  type: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired,
  actions: PropTypes.object,
  isAdmin: PropTypes.bool,
  token: PropTypes.string,
};

Card.defaultProps = {
  isAdmin: false,
};

export default Card;
