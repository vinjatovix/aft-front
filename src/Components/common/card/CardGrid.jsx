import React from "react";
import PropTypes from "prop-types";
import Card from "./Card";

export const CardGrid = ({ data, actions, isAdmin, type, token }) => {
  return (
    <div className="card-grid">
      {data.map((item) => (
        <Card
          type={type}
          key={item._id}
          data={item}
          isAdmin={isAdmin}
          actions={actions}
          token={token}
        />
      ))}
    </div>
  );
};

CardGrid.propTypes = {
  type: PropTypes.string.isRequired,
  data: PropTypes.arrayOf(PropTypes.shape({ _id: PropTypes.string.isRequired }))
    .isRequired,
  actions: PropTypes.object.isRequired,
  isAdmin: PropTypes.bool,
  token: PropTypes.string,
};

CardGrid.defaultProps = {
  isAdmin: false,
};
