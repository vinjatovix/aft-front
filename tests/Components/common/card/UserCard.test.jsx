import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { UserCard } from "../../../../src/Components/common/card/UserCard";
import actions from "../../../fixtures/actions";

describe("UserCard Component", () => {
  const data = {
    username: "user",
    roles: ["aft.admin", "aft.user", "aft.editor"],
  };

  it("should match snapshot", async () => {
    const { container } = render(
      <Router>
        <UserCard data={data} isAdmin={true} actions={actions} token="token" />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });

  it("should show ADMIN", async () => {
    render(
      <Router>
        <UserCard data={data} isAdmin={false} actions={actions} token="token" />
      </Router>
    );
    const li = screen.getByTestId("username");

    expect(li.innerHTML).toContain(data.username);
    expect(screen.getByTestId("role").innerHTML).toContain("ADMIN");
  });

  it("should show EDITOR", async () => {
    render(
      <Router>
        <UserCard
          data={{
            username: "user",
            roles: ["aft.user", "aft.editor"],
          }}
          isAdmin={false}
          actions={actions}
          token="token"
        />
      </Router>
    );
    const li = screen.getByTestId("username");

    expect(li.innerHTML).toContain(data.username);
    expect(screen.getByTestId("role").innerHTML).toContain("EDITOR");
  });

  it("should show USER", async () => {
    render(
      <Router>
        <UserCard
          data={{
            username: "user",
            roles: ["aft.user"],
          }}
          isAdmin={false}
          actions={actions}
          token="token"
        />
      </Router>
    );

    expect(screen.getByTestId("username").innerHTML).toContain(data.username);
    expect(screen.getByTestId("role").innerHTML).toContain("USER");
  });
});
