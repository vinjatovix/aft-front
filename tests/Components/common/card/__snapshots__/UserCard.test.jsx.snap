// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Component should match snapshot 1`] = `
<div>
  <ul
    class="user-list"
  >
    <li
      class=""
      data-testid="username"
    >
      user
    </li>
    <li
      class=""
      data-testid="role"
    >
      <strong
        class="red"
      >
        ADMIN
      </strong>
    </li>
  </ul>
</div>
`;
